packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -1675,12 +1937,28 @@ export class MergeTree {
 			const newSeq = seq === UnassignedSequenceNumber ? Number.MAX_SAFE_INTEGER : seq;
 			const segSeq =
 				node.seq === UnassignedSequenceNumber ? Number.MAX_SAFE_INTEGER - 1 : node.seq ?? 0;
-			return newSeq > segSeq;
+
+			return (
+				newSeq > segSeq ||
+				(node.movedSeq !== undefined &&
+					node.movedSeq !== UnassignedSequenceNumber &&
+					node.movedSeq > seq) ||
+				(node.removedSeq !== undefined &&
+					node.removedSeq !== UnassignedSequenceNumber &&
+					node.removedSeq > seq)
+			);
 		} else {
 			return true;
 		}
 	}
 
+	private getSmallestSeqMoveOp(): number | undefined {
+		return (
+			this.moveSeqs.find((seq) => seq > this.collabWindow.minSeq) ??
```

Comment:
Why is this necessary? `collabWindow.minSeq` should only change via a call to `setMinSeq`, at which point we should evict old entries.